common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,126 @@
+
+    def send(self, event):
+        """
+        Forward the event to the OLI analytics server
+        Exact API here: https://docs.google.com/document/d/1ZB-qwP0bV7ko_xJdJNX1PYKvTyYd4I8CBltfac4dlfw/edit?pli=1#
+        OAuth 1 with nonce and body signing
+        """
+        if not self.oli_analytics_service_url or not self.secret or not self.key:
+            return
+
+        # Only currently passing problem_check events, which are CAPA only
+        if event.get('event_type', '') != 'problem_check':
+            return
+
+        if event.get('event_source', '') != 'server':
+            return
+
+        context = event.get('context')
+        if not context:
+            return
+
+        course_id = context.get('course_id')
+        if not course_id or course_id not in self.course_id_set:
+            return
+
+        user_id = context.get('user_id')
+        if not user_id:
+            return
+
+        event_data = event.get('event')
+        if not event_data:
+            return
+
+        problem_id = event_data.get('problem_id')
+        if not problem_id:
+            return
+
+        success = event_data.get('success')
+        if not success:
+            return
+
+        is_correct = success == 'correct'
+
+        # put the most expensive operation (DB access) at the end, to not do it needlessly
+        try:
+            user = User.objects.get(pk=user_id)
+        except User.DoesNotExist:
+            return
+
+        payload = {
+            'course_id': course_id,
+            'resource_id': problem_id,
+            'student_id': self._get_student_id(user),
+            'result': is_correct
+        }
+
+        request_payload_string = json.dumps({'payload': json.dumps(payload)})
+        request_payload = {'request': request_payload_string}
+        endpoint = self.oli_analytics_service_url + self.path
+
+        try:
+            response = self.oauth.put(endpoint, request_payload)
+
+            # Note: John Orr's API always returns status_code=200, regardless of the actual error,
+            # so response.status_code is misleading.
+            # You should check the actual contents of the response.
+            # The response body will look like {"status":200, "message":"OK", "payload": .....}.
+
+            # Because Google prepends ')]}'\n to all json responses, to protect
+            # against XSSI (cross-site-scripting-inclusion), need to reformat the
+            # returned payloads
+
+            log.info("req.content" + response.content)
+            message = json.loads(response.content.split('\n')[-1])
+            if message['status'] == 200:
+                return "OK"
+            else:
+                return "Error"
+        except Exception:
+            log.warning('Unable to send event to OLI analytics service')
+
+    def _get_student_id(self, user):
+        """
+        Return the anonymous id for a user.
+        """
+        return anonymous_id_for_user(user, None)
```

Comment:
I think None is needed here.  Otherwise I get this : 
anonymous_id_for_user() takes at least 2 arguments (1 given)